# Etude sur le chlorure de vinyle monomère (CVM)


In [2]:
# Nous commencons par importer les librairies nécessaires pour l'analyse des données.
import pandas as pd
import duckdb

con = duckdb.connect(database="./../../database/data.duckdb", read_only=True)

In [3]:
# Affichons les tables

con.table("sise_communes").df()

inseecommune                  nomcommune              quartier  \
0            01001  ABERGEMENT-CLEMENCIAT (L')                     -   
1            01002    ABERGEMENT-DE-VAREY (L')                     -   
2            01004           AMBERIEU-EN-BUGEY             Vareilles   
3            01004           AMBERIEU-EN-BUGEY  St Germain_Brédevent   
4            01004           AMBERIEU-EN-BUGEY                 ville   
...            ...                         ...                   ...   
49550        97617                    TSINGONI      MIRERENI VILLAGE   
49551        97617                    TSINGONI       MROUALE VILLAGE   
49552        97617                    TSINGONI       TSINGONI CENTRE   
49553        97701            SAINT-BARTHELEMY      Toute la commune   
49554        97801                SAINT-MARTIN      Toute la commune   

        cdreseau                       nomreseau  debutalim  
0      001000556        BDS ST DIDIER/CHALARONNE 2010-09-07  
1      001000369           L'ABERGEMENT-DE-VAREY 2010-09-07  
2      001000248              AMBERIEU VAREILLES 2010-09-07  
3      001000249  AMBERIEU SAINT GERMAIN DOUVRES 2010-09-07  
4      001000251                  AMBERIEU VILLE 2010-09-07  
...          ...                             ...        ...  
49550  976000059                MIRERENI COMBANI 2013-05-30  
49551  976000059                MIRERENI COMBANI 2013-05-30  
49552  976000059                MIRERENI COMBANI 2013-05-30  
49553  971000159                SAINT-BARTHELEMY 2010-10-19  
49554  971000160                    SAINT-MARTIN 2010-10-19  

[49555 rows x 6 columns]

In [4]:
# Chargeons la table sise_communes dans un pandas dataframe

communes = con.table("sise_communes").to_df()

In [5]:
con.table("sise_prelevements").df()

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont  \
0         001  001000003             01007        AMBRONAY          None   
1         001  001000003             01007        AMBRONAY          None   
2         001  001000003             01007        AMBRONAY     001001304   
3         001  001000003             01007        AMBRONAY          None   
4         001  001000003             01007        AMBRONAY          None   
...       ...        ...               ...             ...           ...   
404207    976  976003554             97615        PAMANDZI     976000056   
404208    976  976003554             97615        PAMANDZI          None   
404209    976  976003883             97601           ACOUA          None   
404210    976  976003883             97601           ACOUA          None   
404211    976  976003883             97613   M'TSANGAMOUJI          None   

            nomreseauamont pourcentdebit referenceprel   dateprel heureprel  \
0                     None          None   00100139034 2024-01-23     11h27   
1                     None          None   00100139393 2024-02-13     12h49   
2       TTP (CLG) AMBRONAY         100 %   00100139969 2024-03-13     10h46   
3                     None          None   00100140209 2024-04-05     07h27   
4                     None          None   00100140543 2024-05-13     12h43   
...                    ...           ...           ...        ...       ...   
404207         UP PAMANDZI         100 %   97600032142 2024-09-10     09h20   
404208                None          None   97600032321 2024-11-12     10h00   
404209                None          None   97600031330 2024-01-11     09h55   
404210                None          None   97600031370 2024-01-17     11h00   
404211                None          None   97600031371 2024-01-17     10h12   

                                           conclusionprel  \
0       Eau d'alimentation conforme aux limites de qua...   
1       Eau d'alimentation conforme aux exigences de q...   
2       Eau d'alimentation conforme aux exigences de q...   
3       Eau d'alimentation conforme aux exigences de q...   
4       Eau d'alimentation conforme aux exigences de q...   
...                                                   ...   
404207  Eau d'alimentation non conforme aux limites de...   
404208  Eau d'alimentation conforme aux exigences de q...   
404209  Eau d'alimentation conforme aux exigences de q...   
404210  Eau d'alimentation conforme aux exigences de q...   
404211  Eau d'alimentation conforme aux exigences de q...   

                               ugelib                        distrlib  \
0       SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY   
1       SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY   
2       SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY   
3       SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY   
4       SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY   
...                               ...                             ...   
404207                        MAYOTTE                            SMAE   
404208                        MAYOTTE                            SMAE   
404209                        MAYOTTE                            SMAE   
404210                        MAYOTTE                            SMAE   
404211                        MAYOTTE                            SMAE   

                                moalib plvconformitebacterio  \
0       SIE REGION D'AMBERIEU-EN-BUGEY                     C   
1       SIE REGION D'AMBERIEU-EN-BUGEY                     C   
2       SIE REGION D'AMBERIEU-EN-BUGEY                     C   
3       SIE REGION D'AMBERIEU-EN-BUGEY                     C   
4       SIE REGION D'AMBERIEU-EN-BUGEY                     C   
...                                ...                   ...   
404207             LES EAUX DE MAYOTTE                     N   
404208             LES 

In [6]:
# Chargeons la table sise_prelevements dans un pandas dataframe

prelevements = con.table("sise_prelevements").to_df()

In [7]:
con.table("sise_resultats").df().head()

cddept referenceprel cdparametresiseeaux  cdparametre       libmajparametre  \
0    001   00100138503             ALTMICR       1370.0  ALUMINIUM TOTAL µG/L   
1    001   00100138503                 ASP       6489.0   ASPECT (QUALITATIF)   
2    001   00100138503                 ASP       6489.0   ASPECT (QUALITATIF)   
3    001   00100138503              CL2LIB       1398.0          CHLORE LIBRE   
4    001   00100138503              CL2TOT       1399.0          CHLORE TOTAL   

        libminparametre libwebparametre qualitparam insituana          rqana  \
0  Aluminium total µg/l            None           N         L            <10   
1   Aspect (qualitatif)            None           O         L  Aspect normal   
2   Aspect (qualitatif)            None           O         L  Aspect normal   
3          Chlore libre            None           N         T           0,30   
4          Chlore total            None           N         T           0,34   

  cdunitereferencesiseeaux cdunitereference limitequal     refqual  \
0                     µg/L              133       None  <=200 µg/L   
1               SANS OBJET                X       None        None   
2               SANS OBJET                X       None        None   
3                mg(Cl2)/L              165       None        None   
4                mg(Cl2)/L              165       None        None   

   valtraduite   casparam referenceanl  
0         0.00  7429-90-5  00100147233  
1         0.00       None  00100147232  
2         0.00       None  00100147233  
3         0.30       None         None  
4         0.34       None         None

In [8]:
# Chargeons la table sise_resultats dans un pandas dataframe

resultats = con.table("sise_resultats").to_df()

## Combien de relevés sont en dépassement de seuils pour le Chlorure de vinyle monomère (CVM) ?
* On va retenir un seuil de dépassement de 0,5μg/L 

In [9]:
# On va filtrér les prélevements de CVM avec la colonne "libminparametre"

In [10]:
# Pour voir toutes les valeurs dans cette colonne on a fait deux methodes.
noms_prelevements = resultats.libminparametre.unique()
noms_prelevements

array(['Aluminium total µg/l', 'Aspect (qualitatif)', 'Chlore libre', ...,
       'Tétrachlorobenzène-1,2,3,5', 'Acide chlorodibromacétique',
       "Température de mesure de l'oxygène dissous"],
      shape=(1336,), dtype=object)

In [11]:
# Pour voir toutes les valeurs en document appuyer open in a text editor et chercher "Chlorure de vinyle monomère" avec CMD+F
for el in noms_prelevements:
    print(el)

Aluminium total µg/l
Aspect (qualitatif)
Chlore libre
Chlore total
Bioxyde de chlore mg/L ClO2
Couleur (qualitatif)
Fer total
Ozone
Odeur (qualitatif)
pH
Température de l'eau
Bact. et spores sulfito-rédu./100ml
Conductivité à 25°C
Coloration
Bactéries coliformes /100ml-MS
Escherichia coli /100ml - MF
Bact. aér. revivifiables à 22°-68h
Bact. aér. revivifiables à 36°-44h
Ammonium (en NH4)
Saveur (qualitatif)
Entérocoques /100ml-MS
Turbidité néphélométrique NFU
Acénaphtène
Acrylamide
Anthracène
Benzo(a)pyrène *
Benzo(b)fluoranthène
Benzanthracène
Benzo(g,h,i)pérylène
Benzo(k)fluoranthène
Cadmium
Chrysène
Chlorure de vinyl monomère
Chrome total
Cuivre
Dibenzo(a,h)anthracène
Epichlorohydrine
Fluoranthène *
Fluorène
Hydrocarbures polycycliques aromatiques (4 substances)
Indéno(1,2,3-cd)pyrène
Naphtalène
Nickel
Nitrites (en NO2)
Plomb
Phénantrène
Pyrène
Antimoine
Nitrates (en NO3)
Nitrates/50 + Nitrites/3
Manganèse total
Bromoforme
Chloroforme
Chlorodibromométhane
Dichloromonobromométhane
Tri

In [12]:
# Puis on va filtrér ces prélevements

df_cvm = resultats[resultats["libminparametre"] == "Chlorure de vinyl monomère"]
df_cvm

cddept referenceprel cdparametresiseeaux  cdparametre  \
47          001   00100138843               CLVYL       1753.0   
2677        001   00100139016               CLVYL       1753.0   
2724        001   00100139017               CLVYL       1753.0   
2771        001   00100139018               CLVYL       1753.0   
2818        001   00100139019               CLVYL       1753.0   
...         ...           ...                 ...          ...   
11916449    974   97400140604               CLVYL       1753.0   
11916661    974   97400140609               CLVYL       1753.0   
11916873    974   97400140611               CLVYL       1753.0   
11926539    976   97600031907               CLVYL       1753.0   
11926584    976   97600031910               CLVYL       1753.0   

                     libmajparametre             libminparametre  \
47        CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
2677      CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
2724      CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
2771      CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
2818      CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
...                              ...                         ...   
11916449  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
11916661  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
11916873  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
11926539  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
11926584  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   

         libwebparametre qualitparam insituana   rqana  \
47                  None           N         L  <0,004   
2677                None           N         L  <0,004   
2724                None           N         L  <0,004   
2771                None           N         L  <0,004   
2818                None           N         L  <0,004   
...                  ...         ...       ...     ...   
11916449            None           N         L   <0,05   
11916661            None           N         L   <0,05   
11916873            None           N         L   <0,05   
11926539            None           N         L   <0,05   
11926584            None           N         L   <0,05   

         cdunitereferencesiseeaux cdunitereference  limitequal refqual  \
47                           µg/L              133  <=0.5 µg/L    None   
2677                         µg/L              133  <=0.5 µg/L    None   
2724                         µg/L              133  <=0.5 µg/L    None   
2771                         µg/L              133  <=0.5 µg/L    None   
2818                         µg/L              133  <=0.5 µg/L    None   
...                           ...              ...         ...     ...   
11916449                     µg/L              133  <=0.5 µg/L    None   
11916661                     µg/L              133  <=0.5 µg/L    None   
11916873                     µg/L              133  <=0.5 µg/L    None   
11926539                     µg/L              133  <=0.5 µg/L    None   
11926584                     µg/L              133  <=0.5 µg/L    None   

          valtraduite casparam referenceanl  
47                0.0  75-01-4  00100147616  
2677              0.0  75-01-4  00100147791  
2724              0.0  75-01-4  00100147792  
2771              0.0  75-01-4  00100147793  
2818              0.0  75-01-4  00100147794  
...               ...      ...          ...  
11916449          0.0  75-01-4  97400168161  
11916661          0.0  75-01-4  97400168166  
11916873          0.0  75-01-4  97400168168  
11926539          0.0  75-01-4  97600041700  
11926584          0.0  75-01-4  97600041709  

[32293 rows x 17 columns]

In [13]:
# Mainetenant on va être sûr si toutes les unités sont mêmes et oui, ils sont tous µg/L

df_cvm["cdunitereferencesiseeaux"].value_counts()

cdunitereferencesiseeaux
µg/L    32293
Name: count, dtype: int64

In [14]:
# Puis on va filtrér les valeurs supèrieur de la seuil 0.5 dans la colonne "valtraduite"
df_cvm[df_cvm["valtraduite"] > 0.5].head()

cddept referenceprel cdparametresiseeaux  cdparametre  \
110191    001   00100142504               CLVYL       1753.0   
613987    006   00600253912               CLVYL       1753.0   
642000    006   00600255263               CLVYL       1753.0   
642497    006   00600255305               CLVYL       1753.0   
686168    007   00700197770               CLVYL       1753.0   

                   libmajparametre             libminparametre  \
110191  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
613987  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
642000  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
642497  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   
686168  CHLORURE DE VINYL MONOMÈRE  Chlorure de vinyl monomère   

       libwebparametre qualitparam insituana rqana cdunitereferencesiseeaux  \
110191            None           N         L  0,51                     µg/L   
613987            None           N         L  0,55                     µg/L   
642000            None           N         L  0,88                     µg/L   
642497            None           N         L  0,83                     µg/L   
686168            None           N         L  0,74                     µg/L   

       cdunitereference  limitequal refqual  valtraduite casparam referenceanl  
110191              133  <=0.5 µg/L    None         0.51  75-01-4  00100151487  
613987              133  <=0.5 µg/L    None         0.55  75-01-4  00600253922  
642000              133  <=0.5 µg/L    None         0.88  75-01-4  00600255273  
642497              133  <=0.5 µg/L    None         0.83  75-01-4  00600255315  
686168              133  <=0.5 µg/L    None         0.74  75-01-4  00700198485

In [15]:
df_cvm[df_cvm["valtraduite"] > 0.5].shape

(397, 17)

* On a 397 prélevement en dépassement de seuils (0.5 µg/L) pour le Chlorure de vinyle monomère (CVM)

## Combien de communes ont un relevé de dépassement de seuil ?

* Pour cela on va merger les dataframes communes, prelevements et resultats. Il y a documentations en pdf pour voir les jointures sur ce lien https://www.data.gouv.fr/fr/datasets/r/36afc708-42dc-4a89-b039-7fde6bcc83d8

In [16]:
pd.set_option("display.max_columns", None)  # Afficher toutes les colonnes
# pd.set_option('display.width', 1000)  # Ajuster la largeur de l'affichage

In [17]:
# Jointure entre "communes" et "prelevements" sur "cdreseau"
merged_1 = pd.merge(communes, prelevements, on="cdreseau", how="left")

# Jointure entre merged_1 et "resultats" sur referenceprel
final_df = pd.merge(merged_1, resultats, on="referenceprel", how="left")
final_df.head()

inseecommune                  nomcommune quartier   cdreseau  \
0        01001  ABERGEMENT-CLEMENCIAT (L')        -  001000556   
1        01001  ABERGEMENT-CLEMENCIAT (L')        -  001000556   
2        01001  ABERGEMENT-CLEMENCIAT (L')        -  001000556   
3        01001  ABERGEMENT-CLEMENCIAT (L')        -  001000556   
4        01001  ABERGEMENT-CLEMENCIAT (L')        -  001000556   

                  nomreseau  debutalim cddept_x inseecommuneprinc  \
0  BDS ST DIDIER/CHALARONNE 2010-09-07      001             01123   
1  BDS ST DIDIER/CHALARONNE 2010-09-07      001             01123   
2  BDS ST DIDIER/CHALARONNE 2010-09-07      001             01123   
3  BDS ST DIDIER/CHALARONNE 2010-09-07      001             01123   
4  BDS ST DIDIER/CHALARONNE 2010-09-07      001             01123   

         nomcommuneprinc cdreseauamont nomreseauamont pourcentdebit  \
0  CORMORANCHE-SUR-SAONE          None           None          None   
1  CORMORANCHE-SUR-SAONE          None           None          None   
2  CORMORANCHE-SUR-SAONE          None           None          None   
3  CORMORANCHE-SUR-SAONE          None           None          None   
4  CORMORANCHE-SUR-SAONE          None           None          None   

  referenceprel   dateprel heureprel  \
0   00100138978 2024-01-19     11h51   
1   00100138978 2024-01-19     11h51   
2   00100138978 2024-01-19     11h51   
3   00100138978 2024-01-19     11h51   
4   00100138978 2024-01-19     11h51   

                                      conclusionprel  \
0  Eau d'alimentation conforme aux exigences de q...   
1  Eau d'alimentation conforme aux exigences de q...   
2  Eau d'alimentation conforme aux exigences de q...   
3  Eau d'alimentation conforme aux exigences de q...   
4  Eau d'alimentation conforme aux exigences de q...   

                        ugelib                      distrlib  \
0  SI BRESSE DOMBES SAONE SAUR  SAUR FRANCE CHALON-SUR-SAONE   
1  SI BRESSE DOMBES SAONE SAUR  SAUR FRANCE CHALON-SUR-SAONE   
2  SI BRESSE DOMBES SAONE SAUR  SAUR FRANCE CHALON-SUR-SAONE   
3  SI BRESSE DOMBES SAONE SAUR  SAUR FRANCE CHALON-SUR-SAONE   
4  SI BRESSE DOMBES SAONE SAUR  SAUR FRANCE CHALON-SUR-SAONE   

                    moalib plvconformitebacterio plvconformitechimique  \
0  SIE BRESSE DOMBES SAONE                     C                     C   
1  SIE BRESSE DOMBES SAONE                     C                     C   
2  SIE BRESSE DOMBES SAONE                     C                     C   
3  SIE BRESSE DOMBES SAONE                     C                     C   
4  SIE BRESSE DOMBES SAONE                     C                     C   

  plvconformitereferencebact plvconformitereferencechim cddept_y  \
0                          C                          C      001   
1                          C                          C      001   
2                          C                          C      001   
3                          C                          C      001   
4                          C                          C      001   

  cdparametresiseeaux  cdparametre                      libmajparametre  \
0                 ASP       6489.0                  ASPECT (QUALITATIF)   
1                BSIR       1042.0  BACT. ET SPORES SULFITO-RÉDU./100ML   
2               CDT25       1303.0                  CONDUCTIVITÉ À 25°C   
3              CL2LIB       1398.0                         CHLORE LIBRE   
4              CL2TOT       1399.0                         CHLORE TOTAL   

                       libminparametre libwebparametre qualitparam insituana  \
0                  Aspect (qualitatif)            None           O         L   
1  Bact. et spores sulfito-rédu./100ml            None           N         L   
2                  Conductivité à 25°C            None           N         L   
3                         Chlore libre            None           N         T   
4                         Chlore total            None           N         T   

           rqa

In [18]:
final_df.shape

(67837771, 39)

In [19]:
# On va filtrér les prélevements CVM
final_df_cvm = final_df[final_df["libminparametre"] == "Chlorure de vinyl monomère"]
final_df_cvm.shape

(160611, 39)

In [20]:
# Filtrer les dépassements de seuil (> 0.5)
final_df_cvm_seuil = final_df_cvm[final_df_cvm["valtraduite"] > 0.5]

print(
    f"Nombre de prélevement avec un dépassement de seuil : {final_df_cvm_seuil.shape[0]}"
)

Nombre de prélevement avec un dépassement de seuil : 1184


In [21]:
# Identifier les communes qui dépassent le seuil (valtraduite > 0.5)
communes_depassement = final_df_cvm.groupby("inseecommune")["valtraduite"].max() > 0.5
# Nombre de communes dépassant le seuil
nombre_communes_depassement = communes_depassement.sum()

# Afficher le résultat
print(f"Nombre de communes qui dépassent le seuil : {nombre_communes_depassement}")

Nombre de communes qui dépassent le seuil : 639


In [22]:
import folium

ModuleNotFoundError: No module named 'folium'

In [26]:
import folium
import geopandas as gpd
import pandas as pd

# Charger les coordonnées des communes françaises (GeoJSON)
url_geojson = "https://raw.githubusercontent.com/gregoiredavid/france-geojson/master/communes.geojson"
gdf_communes = gpd.read_file(url_geojson)[["code", "nom", "geometry"]]
gdf_communes.rename(columns={"code": "inseecommune"}, inplace=True)

# Filtrer les prélèvements CVM
final_df_cvm = final_df[final_df["libminparametre"] == "Chlorure de vinyl monomère"]

# Identifier les communes qui dépassent le seuil (valtraduite > 0.5)
communes_depassement = final_df_cvm.groupby("inseecommune")["valtraduite"].max() > 0.5

# Fusionner avec le fichier des communes pour avoir les coordonnées géographiques
gdf_final = gdf_communes.merge(
    communes_depassement.reset_index(), on="inseecommune", how="left"
)
gdf_final["valtraduite"] = gdf_final["valtraduite"].fillna(
    False
)  # False = pas de dépassement

# Créer une carte centrée sur la France
m = folium.Map(location=[46.603354, 1.888334], zoom_start=6)

# Ajouter chaque commune sur la carte avec la bonne couleur
for _, row in gdf_final.iterrows():
    if row["geometry"] and not row["geometry"].is_empty:
        lon, lat = row["geometry"].centroid.x, row["geometry"].centroid.y
        color = "red" if row["valtraduite"] else "green"
        folium.CircleMarker(
            location=[lat, lon],
            radius=5,
            color=color,
            fill=True,
            fill_color=color,
            fill_opacity=0.6,
            popup=row["nom"],
        ).add_to(m)

# Afficher la carte
m


ModuleNotFoundError: No module named 'folium'